ci: fix coverage report artifact upload

mas-who opened the pull request:
## Done

Fix issue with coverage report artifact upload failing due to build files for `monaco-editor` containing invalid characters.

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - Ensure CI passes

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://lxd-ui-723.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/webteam/job/start-demo/)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

mas-who commented:
Added some steps to show directory contents within the runner environment. Also updated node to 20 for the coverage job. I think it's still worthwhile to enable manual dispatch for coverage, since it's difficult to replicate the issue, we might need to try a few things and iterate on the main branch

mas-who commented on .github/workflows/coverage.yaml:
```diff
@@ -79,6 +80,12 @@ jobs:
+
       - name: Upload coverage report
         if: always()
         uses: actions/upload-artifact@v4
```
not sure if it really matters since in `pr.yaml` we are using @v4 anyway and it's not failing, but let's try anyway :sweat_smile:

mas-who commented:
just doing a test quick on the fork, will merge after

edlerd reviewed (approved):
LGTM, hope this fixes it :crossed_fingers:

mas-who commented:
> LGTM, hope this fixes it 🤞

I was able to replicate the issue on my fork and this fixed it there, so chance is high but let's see :sweat_smile:
